api/src/routes/protected/challenge.ts

```diff
@@ -776,5 +776,67 @@ export const challengeRoutes: FastifyPluginCallbackTypebox = (
     }
   );
 
+  fastify.post(
+    '/submit-quiz-attempt',
+    {
+      schema: schemas.submitQuizAttempt,
+      errorHandler(error, request, reply) {
+        if (error.validation) {
+          void reply.code(400);
+          void reply.send({
+            type: 'error',
+            message:
+              'That does not appear to be a valid quiz attempt submission.'
+          });
+        } else {
+          fastify.errorHandler(error, request, reply);
+        }
+      }
+    },
+    async req => {
+      const { challengeId, quizId } = req.body;
+
+      const user = await fastify.prisma.user.findUniqueOrThrow({
+        where: { id: req.user?.id },
+        select: {
+          id: true,
+          quizAttempts: true
+        }
+      });
+
+      const existingAttemptIdx = user.quizAttempts.findIndex(
+        attempt =>
+          attempt.challengeId === challengeId && attempt.quizId === quizId
+      );
```

Comment:
```suggestion
      const existingAttempt = user.quizAttempts.find(
        matches({ quizId, challengeId })
      );
```
lodash's matches cleans this up a little, imo. 

The change from findIndex to find is mostly cosmetic. If we use the mongo array manipulation methods (see below!) we only care if we've found something, not which index it is.